### PR TITLE
fix(suite-native): open links via useOpenLink to catch errors

### PR DESCRIPTION
**Pull request by vytick**
Open links via useOpenLink and show toast on failed links.

Resolve [#105](https://github.com/trezor/trezor-suite-private/issues/105)

### CODE REVIEW

File: suite-native/link/src/useOpenLink.ts [hunk 1]
```diff
@@ -6,19 +6,29 @@ import { useToast } from '@suite-native/toasts';
...
+        showToast({
+            variant: 'error',
+            icon: 'warningTriangle',
+            message: 'Unable to open the link',
+        });
+    }, [showToast]);
+
     const handleOpenLink = useCallback(
         async (href: string) => {
+            const canOpenURL = await Linking.canOpenURL(href);
```
Comment (matejkriz) [inline suite-native/link/src/useOpenLink.ts]: I would rather move it inside the `try` block too, even checking if canOpenURL can fail.

**Review by matejkriz — CHANGES_REQUESTED** [review]
Cool, but there are imports `import { Linking } from 'react-native';` no longer used.

File: suite-native/link/src/useOpenLink.ts (hunk 1 above)
Comment (vytick) [reply suite-native/link/src/useOpenLink.ts]: 1f27a23501

**Review by matejkriz — APPROVED** [review]
🎉

**Pull request merged**